src/main/scala/SBTS3Resolver.scala

```diff
@@ -85,7 +86,9 @@ object SbtS3Resolver extends AutoPlugin {
     lazy val s3region       = settingKey[Region]("AWS Region for your S3 resolvers")
     lazy val s3overwrite    = settingKey[Boolean]("Controls whether publishing resolver can overwrite artifacts")
     lazy val s3sse          = settingKey[Boolean]("Controls whether publishing resolver will use server side encryption")
+    @deprecated("Use s3optAcl instead. This currently forwards the value to s3optAcl, but is overwritten by it.")
     lazy val s3acl          = settingKey[S3ACL]("Controls whether published artifacts are accessible publicly via http(s) or not")
+    lazy val s3optAcl       = settingKey[Option[S3ACL]]("Controls published artifacts visibility via http(s) or inherits bucket setting")
```

Comment:
I suggest a bit different approach: keep one key `s3acl`, change its type to `Option[S3ACL]` and add an implicit conversion `S3ACL => Some[S3ACL]`. Then you can _deprecate that implicit conversion_ with an explanation that the type has changed and a suggestion to write `Some(...)` explicitly.

This way things also don't break (from users' point of view) and plus we keep the same setting key.

Reply:
Makes sense. :+1: